reference/tools: add error cases for ET (#1548)
pingcap/docs: dev/reference/tools/error-case-handling/load-misuse-handling.md

---

> title: Common misuses during full data import

Author: Got it, thanks!

category: reference
---

# Common misuses during full data import

This article introduces the common error scenarios caused by using [Loader](/dev/reference/tools/loader.md) or [TiDB Data Migration](/dev/reference/tools/data-migration/overview.md) (DM) to import full data, the causes and solutions of which are also provided.

> ## Error: ``Try adjusting the `max_allowed_packet` variable``

Reviewer: ```suggestion
## Error: ```Try adjusting the `max_allowed_packet` variable```
```

The following error was encountered during full data import.

```
packet for query is too large. Try adjusting the 'max_allowed_packet' variable
```

### Reasons

> * Both MySQL client and MySQL/TiDB Server have `max_allowed_packet` quotas. If any `max_allowed_packet` quota is violated, the client receives a corresponding error message. Currently, the latest version of Syncer, Loader, DM and TiDB Server all have a default `max_allowed_packet` quota of `64M`.

Reviewer: ```suggestion
* Both MySQL client and MySQL/TiDB Server have `max_allowed_packet` quotas. If any of the `max_allowed_packet` quotas is violated, the client receives a corresponding error message. Currently, the latest version of Syncer, Loader, DM, and TiDB Server all have a default `max_allowed_packet` quota of `64M`.
```

    * Please use the latest version, or the latest stable version of the tool. See the [download page](/dev/reference/tools/download.md).

> * The full data import processing module in Loader or DM Loader does not support splitting `dump sqls` files. It is because Mydumper uses the simplest code to implement, as stated in the comment `/* Poor man's data dump code */`. If you split files in Loader or DM, you need to implement a complete parser based on `TiDB parser` to handle data splits correctly, which causes the following problems:

Reviewer: ```suggestion
* The full data import processing module in Loader or DM does not support splitting `dump sqls` files. It is because Mydumper uses the simplest code to implement, as stated in the comment `/* Poor man's data dump code */`. If you split files in Loader or DM, you need to implement a complete parser based on `TiDB parser` to handle data splits correctly, which causes the following problems:
```

    * High workload
    * High complexity with poor correctness
    * Performance drop

### Solutions

> * For the above reasons, it is recommended to execute `-s, --statement-size` statement which Mydumper offered to control the size of `Insert Statement`: `Attempted size of INSERT statement in bytes, default 1000000`.

Reviewer: ```suggestion
* For the above reasons, it is recommended to use the `-s, --statement-size` option which Mydumper offers to control the size of `Insert Statement`: `Attempted size of INSERT statement in bytes, default 1000000`.
```

>     According to the default configuration of `--statement-size`, the size of `Insert Statement` is as close as `1M`. The default configuration ensures this problem will not occur in most cases.

Reviewer: ```suggestion
    According to the default configuration of `--statement-size`, Mydumper generates the `Insert Statement` whose size is as close as `1M`. This default configuration ensures that this error will not occur in most cases.
```

>     Sometimes the following `WARN` log will appear during dump to indicate that the table of dump may be a wide table. The `WARN` log itself will not affect dump process.

Reviewer: ```suggestion
    Sometimes the following `WARN` log appears during the dump process. The `WARN` log itself does not affect the dump process but indicates that the dumped table might be a wide table.
```

    ```
    Row bigger than statement_size for xxx
    ```

> * If a single row of a wide table exceeds 64M, the following two configurations need to be modified and enabled.

Reviewer: ```suggestion
* If a single row of a wide table exceeds 64M, you need to modify and enable the following two configurations.
```
